app/src/main/java/org/covidwatch/android/ui/home/HomeFragment.kt

```diff
@@ -1,76 +1,36 @@
-import androidx.fragment.app.Fragment
 import androidx.lifecycle.Observer
 import androidx.navigation.fragment.findNavController
-import org.covidwatch.android.BuildConfig
-import org.covidwatch.android.R
-import org.covidwatch.android.data.FirstTimeUser
-import org.covidwatch.android.data.ReturnUser
-import org.covidwatch.android.data.Setup
+import org.covidwatch.android.*
+import org.covidwatch.android.data.CovidExposureSummary
 import org.covidwatch.android.databinding.FragmentHomeBinding
+import org.covidwatch.android.ui.BaseFragment
+import org.covidwatch.android.ui.event.EventObserver
 import org.koin.androidx.viewmodel.ext.android.viewModel
-import org.covidwatch.android.getFirebaseId
-import org.covidwatch.android.setTester
-import org.covidwatch.android.setAnalyticsInstanceFromContext
-import org.covidwatch.android.sendEvent
-import android.content.ComponentName
-import android.content.pm.ActivityInfo
-import android.content.pm.PackageManager
-import android.os.Bundle
-import org.covidwatch.android.ui.MainActivity
 
-class HomeFragment : Fragment() {
-
-    private var _binding: FragmentHomeBinding? = null
-    private val binding get() = _binding!!
+class HomeFragment : BaseFragment<FragmentHomeBinding>() {
 
     private val homeViewModel: HomeViewModel by viewModel()
 
-    override fun onCreateView(
-        inflater: LayoutInflater,
-        container: ViewGroup?,
-        savedInstanceState: Bundle?
-    ): View? {
-        _binding = FragmentHomeBinding.inflate(inflater, container, false)
-        return binding.root
-    }
-
-    override fun onDestroyView() {
-        super.onDestroyView()
-        _binding = null
-    }
+    override fun binding(inflater: LayoutInflater, container: ViewGroup?): FragmentHomeBinding =
+        FragmentHomeBinding.inflate(inflater, container, false)
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
         //Add meta-data test here
         getFirebaseIdIfTester()
-        binding.swipeRefreshLayout.setColorSchemeColors(
-            ContextCompat.getColor(
-                requireContext(),
-                R.color.colorPrimary
-            )
-        )
         homeViewModel.onStart()
-        homeViewModel.userFlow.observe(viewLifecycleOwner, Observer { userFlow ->
-            when (userFlow) {
-                is FirstTimeUser -> {
-                    updateUiForFirstTimeUser()
-                }
-                is Setup -> {
-                    findNavController().navigate(R.id.splashFragment)
-                }
-                is ReturnUser -> {
-                    updateUiForReturnUser()
-                }
-            }
+        homeViewModel.navigateToOnboardingEvent.observe(viewLifecycleOwner, EventObserver {
+            findNavController().navigate(R.id.splashFragment)
         })
+        homeViewModel.exposureSummary.observe(viewLifecycleOwner, Observer(::bindExposureSummary))
```

Comment:
@madim I forgot to comment that.
Please, use in the future `LifecycleOwner` extensions. `observe` and `observeEvent`
```
with(viewModel) {
    observe(data) {
        ...
    }
    observeEven(event) { 
      ...
    }
}
```